pennylane/pauli/pauli_word_sparse_matrix.md

```diff
@@ -0,0 +1,119 @@
+
+In this case, $Z$ extends $\texttt{val}$ by replicating it with a negative sign. $\texttt{col}$ is extended by replicating it and adding the dimension $m$ to values (see below for a summary of matrix effects).
+
+Now if we apply $X$, we obtain:
+$$ XZI = \begin{pmatrix}0 & 1\\ 1 & 0\end{pmatrix}\otimes 
+\begin{pmatrix}
+1 & 0 & 0 & 0 \\ 
+0 & 1 & 0 & 0 \\ 
+0 & 0 & -1 & 0 \\
+0 & 0 & 0 & -1
+\end{pmatrix} 
+= 
+\left(\begin{array}{cccc|cccc}
+& & & & 1 & 0 & 0 & 0 \\ 
+& & & & 0 & 1 & 0 & 0 \\
+& & & & 0 & 0 & -1 & 0 \\
+& & & & 0 & 0 & 0 & -1 \\
+\hline 
+1 & 0 & 0 & 0 & & & & \\
+0 & 1 & 0 & 0 & & & & \\
+0 & 0 & -1 & 0 & & & & \\
+0 & 0 & 0 & -1 & & & & \\
+\end{array}\right).$$
+
+The new CSR representation is $\texttt{val}=\left[1, 1, -1, -1, 1, 1, -1, -1\right]$ and $\texttt{col}=\left[4, 5, 6, 7, 0, 1, 2, 3\right]$.
+In this case, $X$ has simply duplicated the $\texttt{val}$, and then we replicate $\texttt{col}$ and add the dimension $m$ to the first indices.
+
+```python
+XZI = np.kron(X, ZI)
+XZI_CSR = scipy.sparse.csr_matrix(XZI)
+print(XZI)
+print(f"val: {XZI_CSR[XZI_CSR.nonzero()].tolist()[0]}\ncol: {XZI_CSR.indices}")
+
+[[ 0  0  0  0  1  0  0  0]
+ [ 0  0  0  0  0  1  0  0]
+ [ 0  0  0  0  0  0 -1  0]
+ [ 0  0  0  0  0  0  0 -1]
+ [ 1  0  0  0  0  0  0  0]
+ [ 0  1  0  0  0  0  0  0]
+ [ 0  0 -1  0  0  0  0  0]
+ [ 0  0  0 -1  0  0  0  0]]
+val: [1, 1, -1, -1, 1, 1, -1, -1]
+col: [4 5 6 7 0 1 2 3]
+```
+
+In general, we take our current $\texttt{val}$ and replicate it multiplying by the Pauli operator's $\texttt{val}_P$, such that $\texttt{val}\leftarrow[\texttt{val}, \texttt{val}]\cdot\texttt{val}_P$ (assuming $\texttt{val}_P$ broadcasts accordingly).
+For the $\texttt{col}$, we replicate it and add the previous matrix dimension $m$ times the operator's $\texttt{col}_P$, such that $\texttt{col}\leftarrow[\texttt{col}, \texttt{col}] + m\times\texttt{col}_P$.
+Knowing the number of terms $n$, we can directly compute the resulting $\texttt{row}$ as $\texttt{row} = [0, 1, \dots, 2^n]$.
+Below, we list the explicit effect on the CSR representation of performing the tensor product with each Pauli matrix:
+
+$$\begin{align*}
+I &= \begin{cases}\texttt{val} \leftarrow [\texttt{val}, \texttt{val}]\\ \texttt{col}\leftarrow [\texttt{col}, \texttt{col}+m]\end{cases} \\
+X &= \begin{cases}\texttt{val} \leftarrow [\texttt{val}, \texttt{val}]\\ \texttt{col}\leftarrow [\texttt{col}+m, \texttt{col}]\end{cases} \\
+Y &= \begin{cases}\texttt{val} \leftarrow [-i\texttt{val}, i\texttt{val}]\\ \texttt{col}\leftarrow [\texttt{col}+m, \texttt{col}]\end{cases} \\
+Z &= \begin{cases}\texttt{val} \leftarrow [\texttt{val}, -\texttt{val}]\\ \texttt{col}\leftarrow [\texttt{col}, \texttt{col}+m]\end{cases}
+\end{align*}$$ 
```

Comment:
We could also mention that we directly assign the numpy arrays to the attributes of an empty CSR matrix to avoid Scipy's checks and copies in `__init__`.

Reply:
That could go in a comment in the code too. Nice works @BorjaRequena !

Reply:
I put it as a comment in the code. I think it is enough to understand the couple of lines 😀